src/Illuminate/Database/Query/Grammars/Grammar.php

```diff
@@ -9,7 +9,7 @@
 use Illuminate\Support\Str;
 use RuntimeException;
 
-class Grammar extends BaseGrammar
+abstract class Grammar extends BaseGrammar
```

Comment:
Is this necessary for this PR?

Reply:
The `BaseGrammar` (which is `abstract`) is now requiring extending classes to implement the `abstract public function getDriverName();` method. As the query grammar can't implement this for the dbms specific grammars I changed it to be abstract too. Except from unit tests this class is also never initiated once, so it's basically `abstract` by intention?